Source/GTUpdateHelper.swift

```diff
@@ -0,0 +1,183 @@
+//
+//  GTUpdateHelper.swift
+//  GTConstants
+//
+//  Created by Ali Ajmine on 3/2/17.
+//  Copyright © 2017 Mike Silvis. All rights reserved.
+//
+
+import UIKit
+
+public enum Restriction: Int {
+    case low
+    case medium
+    case high
+}
+
+public enum UpdateType {
+    case OS
+    case app
+}
+
+public struct UpdateItem {
+    public let type: UpdateType
+    public let version: String
+    public let restriction: Restriction
+    public let message: String?
+}
+
+internal class GTUpdateHelper {
+
+    fileprivate var appVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    fileprivate var osVersion: String {
+        return UIDevice.current.systemVersion
+    }
+
+    fileprivate struct UpdateConfig {
+
+        struct Version {
+            let number: String
+            let message: String?
+
+            init(number: String, message: String? = nil) {
+                self.number = number
+                self.message = message
+            }
+        }
+
+        let active: Bool
+        let restriction: Restriction
+        let minimumAppVersion: Version
+        let minimumOSVersion: Version
+
+        init(active: Bool, restriction: Restriction, minimumAppVersion: Version, minimumOSVersion: Version) {
+            self.active = active
+            self.restriction = restriction
+            self.minimumAppVersion = minimumAppVersion
+            self.minimumOSVersion = minimumOSVersion
+        }
+    }
+
+    fileprivate var updateConfigs: [UpdateConfig] = []
+
+    //
+    // MARK: - Helper Interface
+    //
+
+    fileprivate(set) internal var updateItem: UpdateItem?
+
+    internal var updateNeeded: Bool {
+        return updateItem != nil
+    }
+
+    internal func configureUpdateRequirements(withData data: AnyObject?) {
+        buildUpdateConfigs(withData: data)
+        configureUpdateRequirements()
+    }
+
+    fileprivate func buildUpdateConfigs(withData data: AnyObject?) {
+        guard let configDicts = data as? [[String: AnyObject]] else { return }
+
+        for dict in configDicts {
```

Comment:
@mattbanach ended up leaving this as a for in since I couldn't do `continue` in forEach